Second stimulus merge

micahwoodard opened the pull request:
### Pull Request instructions:
- Please follow the [update protocol](https://github.com/AllenNeuralDynamics/aind-behavior-blog/wiki/Software-Update-Procedures)
- Answer the questions below in detail. Your responses will be emailed to experimenters. 
- If the experimenters must do anything new, provide detailed step by step instructions on the [wiki](https://github.com/AllenNeuralDynamics/aind-behavior-blog/wiki)
- If computer maintainers need to manually update anything, provide detailed step by step instructions
- Use markdown syntax in order for your comments to be rendered reliably in the email: "1." instead of "1)", use four spaces for indents.
- If you use the keyword "skip email" in the title, it will skip the email updates
- Merges from "develop" into "production_testing" should use the keyword "production merge" in the title for reliable indexing of updates
- Merges from "production_testing" into "main" should use the keyword "update main"
  
### Describe changes:
- updates branch ading_second_stimulus branch with develop 
### What issues or discussions does this update address?

### Describe the expected change in behavior from the perspective of the experimenter
- none
### Describe any manual update steps for task computers
- none
### Was this update tested in 446/447?

### Does this update impact downstream processing by adding new saved files, or changing their format? If so, have you documented changes?

XX-Yin committed:
adding second stimulus to the bonsai workflow

XX-Yin committed:
adding give second stimulus panel

XX-Yin committed:
adding GiveSecondStimulus and SecondStimulusDelay

XX-Yin committed:
sending the second delay stimulus

XX-Yin committed:
changing the reward delay time

XX-Yin committed:
typo and renaming

XX-Yin committed:
typo

XX-Yin committed:
show camera recording

XX-Yin committed:
receiving and saving the second sound

XX-Yin committed:
changing name and logic of soundcard_eventN

XX-Yin committed:
typo

XX-Yin committed:
move position

XX-Yin committed:
move to the audio

XX-Yin committed:
continue

XX-Yin committed:
comment

XX-Yin committed:
changing the name

XX-Yin committed:
adding delay

XX-Yin committed:
adjusting the reward consume time

XX-Yin committed:
changing to float

XX-Yin committed:
adding no reward

XX-Yin committed:
making the reward delay as a distribution

XX-Yin committed:
saving the reward delay and the second stimulus delay

XX-Yin committed:
changing the ephys gui

XX-Yin committed:
removing a label

XX-Yin committed:
adding take one

XX-Yin committed:
changing the name RewardDelay to SecondStimulusDelay

XX-Yin committed:
removing testing line

XX-Yin committed:
adding independent randomness

XX-Yin committed:
adding independent randomness

XX-Yin committed:
changing the name

XX-Yin committed:
moving the position

XX-Yin committed:
updating the ephys gui

XX-Yin committed:
set default as even

micahwoodard committed:
merges second stimulus branch

micahwoodard committed:
fixes bonsai merge errors

micahwoodard committed:
fixes bonasi workflow

micahwoodard committed:
fixes bonasi workflow

micahwoodard committed:
reverts layout

micahwoodard committed:
merges with develop

micahwoodard committed:
adds ui elements back

hagikent commented:
Thanks @micahwoodard . I will test the `second-stimulus-merge` branch on the rig. And get back to you. 

While solving the conflict, any specific non-sencond-stimulus function that should be explicitly tested?

I guess I can close this PR as redundant?  https://github.com/AllenNeuralDynamics/dynamic-foraging-task/pull/1619

alexpiet commented:
@hagikent is there any extra information that needs to be added to the nwbs?

hagikent commented:
@alexpiet thanks for the heads up. should metadata/behavior-json be also modified in a coordinated manner?

rachelstephlee commented:
This was not processed with the correct aind-bonsai-behavior-nwb package. The CS+/CS- is not being packaged in the NWB. Will follow up with Ahad and Kenta separately.

hagikent committed:
adding Micah's Hotfix

There have been errors in fetching curriculus/protocol from metadata. Micah's hotfix targeting the main branch is applied here too.

rachelstephlee commented:
We are waiting for a job_type so we can trigger the right pipeline; that is waiting on me to finish validating one last PR...

micahwoodard commented:
> One question, can we switch job_type (=parameters to be fed into watch-dog) flexibly session-by-session within the same branch...? (if not we cannot merge this into main)

@alexpiet is the job_type currently being set by the setting file under the transfer_service_job_type key? That would mean a user would have to update the settings file which does not seem like a good solution

alexpiet commented:
> > One question, can we switch job_type (=parameters to be fed into watch-dog) flexibly session-by-session within the same branch...? (if not we cannot merge this into main)
> 
> @alexpiet is the job_type currently being set by the setting file under the transfer_service_job_type key? That would mean a user would have to update the settings file which does not seem like a good solution

Yes, its set right now via the settings file. I think its a very bad idea to have a user modify the settings file to run a specific mouse. We could move the job type to the curriculum, or just set logic that determines what the job type is based on task or something.

Why does the second stimulus need a different job type?

Ahad-Allen commented:
That is correct. We are just waiting for Rachel's validation and then we can mark the current PR for the DF + Fiber pipeline as a release that can be triggered with the job type for older sessions. We will then move over to using the fiber + DF split pipelines.
